app/routers/auth.py

```diff
@@ -0,0 +1,133 @@
+from app.middleware.auth import get_current_active_user
+from app.schemas.auth import AuthTokenResponse, UserFromDB, UserLogin, UserRegister
+from app.database import get_db
+import app.services.auth as auth_service
+from uuid import UUID
+
+router = APIRouter(
+    prefix="/auth",
+    tags=["authentication"],
+    responses={
+        status.HTTP_404_NOT_FOUND: {"description": "Not found"}
+    },
+)
+
+
+def check_user_found(user: UserFromDB) -> UserFromDB:
+    if user is None:
+        raise HTTPException(
+            status_code=status.HTTP_404_NOT_FOUND,
+            detail="User not found"
+        )
+    return user
+
+
+@router.post("/register", response_model=UserFromDB)
+async def register(user: UserLogin, db: Session = Depends(get_db)):
+    try:
+        return auth_service.register(UserRegister.model_validate(user.model_dump()), db=db)
+    except Exception as e:
+        raise HTTPException(
+            status_code=status.HTTP_400_BAD_REQUEST,
+            detail=str(e)
+        )
+
+
+@router.post("/login", response_model=AuthTokenResponse)
+async def login(
+    form_data: Annotated[OAuth2PasswordRequestForm, Depends()],
+    db: Session = Depends(get_db)
+):
+    try:
+        return auth_service.login(form_data.username, form_data.password, form_data.scopes, db=db)
+    except Exception as e:
+        raise HTTPException(
+            status_code=status.HTTP_400_BAD_REQUEST,
+            detail=str(e)
+        )
+
+
+@router.get(
+    "/users/me",
+    response_model=UserFromDB,
+    description="Get info about logged in user. Requires 'me:read' scope.",
+)
+async def read_users_me(
+    current_user: Annotated[UserFromDB, Security(get_current_active_user, scopes=["me:read"])],
+):
+    return current_user
+
+
+@router.get(
+    "/users/all",
+    response_model=list[UserFromDB],
+    dependencies=[Security(
+        get_current_active_user,
+        scopes=["users:read"]
+    )],
+    description="Get info about all users. Requires 'users:read' scope.",
+)
+async def read_all_users(db: Session = Depends(get_db)):
+    return auth_service.get_all(db)
+
+
+@router.get(
+    "/user/{id}",
+    response_model=UserFromDB,
+    dependencies=[Security(
+        get_current_active_user,
+        scopes=["users:read"]
+    )],
+    description="Get info about user by ID. Requires 'user:read' scope.",
+)
+async def read_user_by_id(id: UUID, db: Session = Depends(get_db)):
+    return check_user_found(auth_service.get_by_id(id, db))
+
+
+@router.get(
+    "/user/by-username/{username}",
+    response_model=UserFromDB,
+    dependencies=[Security(
+        get_current_active_user,
+        scopes=["users:read"]
+    )],
+    description="Get info about user by username. Requires 'user:read' scope.",
+)
+async def read_user_by_username(username: str, db: Session = Depends(get_db)):
+    return check_user_found(auth_service.get_by_username(username, db))
+
+
+@router.put(
+    "/user/",
```

Comment:
```suggestion
    "/users/{id}",
```

V body mají být data, a zdroj má být identifikovaný v cestě.
Chápu, že takhle je to jednodušší, i se to tak někde používá, ale z pohledu principu RESTu, je to špatně.